crates/trie/src/walker.rs

```diff
@@ -184,10 +184,13 @@ impl<C: TrieCursor> TrieWalker<C> {
         }
 
         // Find the next sibling with state.
-        while subnode.nibble() < 16 {
+        loop {
             if subnode.state_flag() {
                 return Ok(())
             }
+            if subnode.nibble() == 0xf {
+                break;
+            }
```

Comment:
this change should be fine because the current subnode gets discarded below anyway with `self.stack.pop()`